From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct inet_peer
Patch-mainline: never, kabi
References: kabi

In 4.4.176, commit 01fb36063d972e1b1ac751d9249300098ba20794 (net:
ipv4: use a dedicated counter for icmp_v4 redirect packets), upstream
commit c09551c6ff7fe16a79a42133bcecba5fc2fc3291 added n_redirects to
struct inet_peer. It made the kABI checker to complain.

Given struct inet_peer is allocated in inet_getpeer from the
peer_cachep pool, let's move the member to the end of the structure
and pretend the change did not happen.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/inetpeer.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/net/inetpeer.h
+++ b/include/net/inetpeer.h
@@ -40,7 +40,6 @@ struct inet_peer {
 
 	u32			metrics[RTAX_MAX];
 	u32			rate_tokens;	/* rate limiting for ICMP */
-	u32			n_redirects;
 	unsigned long		rate_last;
 	union {
 		struct list_head	gc_list;
@@ -62,6 +61,9 @@ struct inet_peer {
 	/* following fields might be frequently dirtied */
 	__u32			dtime;	/* the time of last use of not referenced entries */
 	atomic_t		refcnt;
+#ifndef __GENKSYMS__
+	u32			n_redirects;
+#endif
 };
 
 struct inet_peer_base {
